### DISCUSSION TITLE
blog/user-management-in-aspnet-core-mvc/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#6] [General]
# blog/user-management-in-aspnet-core-mvc/

In this article, let's go in-depth and understand the functionalities you can achieve with the help of Microsoft Identity. We will build a small yet practical implementation of Custom User Management in ASP.NET Core MVC with Identity. This will cover most of the practical use cases involved while developing User Management in ASP.NET Core.

https://codewithmukesh.com/blog/user-management-in-aspnet-core-mvc/

**Comment by Piyush8456** [comment]
here i cannot get login... why?
i follow all your step by step properly from this documenet
after that i cannot login and when i click on login button after fill the username and password  i directly rediret on home page

**Comment by gendonghuang** [comment]
How to go to the “Manage your account” page? What is the web address?
Thanks

**Comment by StevenKlapNL** [comment]
Hi!

Trying to follow this step by step tutorial, however got stuck very early on.. 

Im using .NET 8 and from what I understand, as of .NET 6 there is no more startup.cs as this is now merged in program.cs. 
Under "Adding Custom Fields to Identity User" you advise to add services code to startup.cs, 

my question is how to deal with that in program.cs?

**Reply by iammukeshm** [reply]
Well, you can use `Program.cs` instead. Just work with the `builder.Services.Add..` extension.

**Comment by JiaXiongZeng** [comment]
This article is very helpful and epic.
Could you provide some examples for multiple identity provider support simultaneously?
Using ASP.NET core to build up a log in page, users can log in by the method locally as the example you provide
or log in by some identity server via OIDC like Keycloak, Google, or Facebook......etc.

**Comment by CDian72** [comment]
Congratulation Mukesh. You write awesome clear posts. Very good !!!

**Comment by worbarry** [comment]
i already have a dot net api that handles all the auth how do i scaffold mvc identity pages to use my existing api and not create a  new one?

**Comment by YashSmithh** [comment]
The Repository Pattern in ASP.NET Core simplifies data management, enhances code reusability, and ensures clean architecture. It acts as a bridge between business logic and data access layers, promoting separation of concerns. Developers use repositories to handle CRUD operations efficiently while maintaining testable, maintainable code.
